### PR TITLE
FEAT: addNotes

### PR DESCRIPTION
**Pull request by Adarsh-Patel-Dev**

### CODE REVIEW

File: src/Context/archiveContext.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+// import { useReducer, useContext, createContext} from 'react';
+// import { useNoteContext } from './noteContext';
+
+
+// const ArchiveContext = createContext();
+// const useArchiveContext = ()=> useContext(ArchiveContext);
+
+// const { noteDispatch } = useNoteContext();
+
+// function archiveReducer( state, action ) {
```
Comment (srejitk) [inline src/Context/archiveContext.js:12]: What's all this commented code?

File: src/Context/noteContext.js [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+  async function editNote(e){
+    e.preventDefault();
+    const note = {
+      // _id,
+      title,
+      priority,
+      label,
+      textareaValue,
+      notesBgColor,
+      noteCreatedDate,
```
Comment (srejitk) [inline src/Context/noteContext.js:190]: Remove the console logs

**Review by SinchanaMS — APPROVED** [review]
Good to go once you remove the logs.

File: src/Context/noteContext.js [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+            url: "/api/archives/",
+            headers: { authorization: localStorage.getItem("token") },
+        });
+        if( true ){
+            noteDispatch({ type:"GET_ARCHIVE_NOTES", payload: response.data.archives })
+            console.log(response.data.archives)
+        }
+    } catch(error){
+        console.log(error);
+    }
```
Comment (srejitk) [inline src/Context/noteContext.js:229]: Here too

File: src/Context/noteContext.js [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+  }
+
+
+
+  async function getArchiveNotes(noteDispatch){
+    try{
+        const response = await axios({
+            method: "GET",
+            url: "/api/archives/",
+            headers: { authorization: localStorage.getItem("token") },
```
Comment (srejitk) [inline src/Context/noteContext.js:221]: Here tooo

File: src/Context/trashContext.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
+// import axios from 'axios';
+// import React from 'react'
+// import { useReducer, useContext, createContext} from 'react';
+// import { useNoteContext } from './noteContext';
+
+
```
Comment (srejitk) [inline src/Context/trashContext.js:6]: Remove if not needed

**Review by srejitk — APPROVED** [review]
Looks good, but please remove the logs

File: src/Context/archiveContext.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+//     switch(action.type){
+//         case "GET_ARCHIVE_NOTES":
+//             return { ...state, archiveNotes: action.payload }
```
Comment (Dev-AkshatGupta) [inline src/Context/archiveContext.js:15]: Suggestion: Remove this code its of no use or mention todo comment on top.

File: src/Context/trashContext.js [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+// const TrashContext = createContext();
+// const useTrashContext = ()=> useContext(TrashContext);
+
+// const { noteDispatch } = useNoteContext();
+
+// function trashReducer( state, action ) {
+//     switch(action.type){
+//         case "GET_TRASH_NOTES":
```
Comment (Dev-AkshatGupta) [inline src/Context/trashContext.js:14]: Suggestion:  Remove this code too if its of no use or do the same as I asked earlier

**Review by Dev-AkshatGupta — APPROVED** [review]
Else everything seems good to me you can go ahead and merge it

**Comment by Adarsh-Patel-Dev** [comment]
FEAT: 
- add a new note
- add to archive/unarchive
- add to delete/trash
- edit note
- search by title/description/ label/ priority

**Pull request merged**